mono/metadata/class.c

```diff
@@ -2603,11 +2603,8 @@ mono_type_get_checked (MonoImage *image, guint32 type_token, MonoGenericContext
 	if ((type_token & 0xff000000) != MONO_TOKEN_TYPE_SPEC) {
 		MonoClass *klass = mono_class_get_checked (image, type_token, error);
 
-		if (!klass) {
+		if (!klass || m_class_has_failure (klass))
```

Comment:
I assume this is needed because mono_class_get_checked won't return NULL for broken classes.

This change breaks the invariant that mono_type_get_checked will set an error if it returns NULL, we should avoid this as much as possible and explicit document those cases when unavoidable.